### PR TITLE
Laboratory work #2, Semenova Taisiya - 23FPL2

### PR DESCRIPTION
**Pull request by semtassta**
Documents Relevance Assessment Using BM25

### CODE REVIEW

File: lab_2_retrieval_w_bm25/settings.json [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 6
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/settings.json:2]: good. Let's try for 8

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/settings.json [hunk 2]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 8
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/settings.json:2]: Actually, I do not have comments. What about 10?

File: lab_2_retrieval_w_bm25/settings.json (hunk 1 above)
Comment (semtassta) [reply lab_2_retrieval_w_bm25/settings.json]: done

File: lab_2_retrieval_w_bm25/settings.json (hunk 2 above)
Comment (semtassta) [reply lab_2_retrieval_w_bm25/settings.json]: done

File: lab_2_retrieval_w_bm25/main.py [hunk 3]
```diff
@@ -336,35 +365,37 @@ def load_index(file_path: str) -> list[dict[str, float]] | None:
...
+        loaded_index = load(file)
+    if isinstance(loaded_index, list):
+        return loaded_index
+    return None
 
 
 def calculate_spearman(rank: list[int], golden_rank: list[int]) -> float | None:
     """
-    Calculate Spearman's rank correlation coefficient between two rankings.
+    Calculate Superman's rank correlation coefficient between two rankings.
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:377]: BATMAN 🦇

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
My favorite superhero is Batman. It is a problem for you. Penalty for you is -5 (joke)

File: lab_2_retrieval_w_bm25/main.py (hunk 3 above)
Comment (semtassta) [reply lab_2_retrieval_w_bm25/main.py]: done

File: lab_2_retrieval_w_bm25/main.py [hunk 4]
```diff
@@ -3,10 +3,11 @@
 
 Text retrieval with BM25
 """
-# pylint:disable=too-many-arguments, unused-argument
 from json import dump, load
 from math import log
 
+# pylint:disable=too-many-arguments, unused-argument
```
[the remark this answers is not in the document]
Comment (semtassta) [reply lab_2_retrieval_w_bm25/main.py]: done

**Pull request closed without merging**